src/graphql.tsx

```diff
@@ -320,18 +316,15 @@ export default function graphql(
       }
 
       setInitialProps() {
-        if (this.type === DocumentType.Mutation) return this.createWrappedMutation(this.props);
+        if (this.type === DocumentType.Mutation) {
+          return;
+        }
 
         // Create the observable but don't subscribe yet. The query won't
         // fire until we do.
         const opts: QueryOptions = this.calculateOptions(this.props);
 
-        if (opts.skip) {
-          this.data = assign({}, skippedQueryData) as any;
-        } else {
-          this.data = assign({}, defaultQueryData) as any;
-          this.createQuery(opts);
-        }
+        this.createQuery(opts);
```

Comment:
I think we are OK now because we don't even get to `setInitialProps` if we are skipping: https://github.com/apollostack/react-apollo/blob/bd2f887a559c2fe1ba7b3b89e9b337e3223a9402/src/graphql.tsx#L261